src/components/user/repo/db/trashbin.schema.js

```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const schema = new Schema(
+	{
+		userId: {
+			type: Schema.Types.ObjectId,
+			ref: 'user',
+			required: true,
+			index: true,
+		},
+		user: { type: Object },
```

Comment:
With the current model it is more type safe. The developer has to add additional properties if he wants to add more collections to the trashbin. It will be easier to retrieve the data from the current structure.

Reply:
True, but then we need 2 sepeate handler to add data to database. One for user and account and one for all other stuff that need a generic solution to add all like errors, context, partial deletions, full deletions, context information over external thinks.

The same for restore or shd dashboard read access.